ui/src/BaseEditor.ts

```diff
@@ -0,0 +1,166 @@
+    public abstract getFilename(context: ContextT): Promise<string>;
+    public abstract getSize(context: ContextT): Promise<number>;
+    public abstract getSaveConfirmationText(context: ContextT): Promise<string>;
+
+    public async showEditor(context: ContextT): Promise<void> {
+        const size: number = await this.getSize(context);
+        const fileName: string = await this.getFilename(context);
+        const splitFileName: string[] = fileName.split('.');
+        const extension: string = splitFileName[splitFileName.length - 1];
+
+        this.appendToOutput(localize('opening', 'Opening "{0}" ...', fileName));
+        if (size > 50 /*Megabytes*/) {
+            const message: string = localize('tooLargeError', '"{0}" is too large to download.', fileName);
+
+            await vscode.window.showWarningMessage(message, dialogResponses.OK);
+            this.appendLineToOutput(localize('failed', "Failed."));
+            this.appendLineToOutput(localize('errorDetails', 'Error Details: {0}', message));
+        } else if (extension === 'exe' || extension === 'img' || extension === 'zip') {
+            const message: string = localize('unsupportedError', '"{0}" has an unsupported file extension.', fileName);
+
+            await vscode.window.showWarningMessage(message, dialogResponses.OK);
+            this.appendLineToOutput(localize('failed', " Failed."));
+            this.appendLineToOutput(localize('errorDetails', 'Error Details: {0}', message));
+        } else {
+            try {
+                // tslint:disable-next-line:no-unsafe-any
+                const localFilePath: string = await TemporaryFile.create(fileName);
+                const document: vscode.TextDocument = await vscode.workspace.openTextDocument(localFilePath);
+                this.fileMap[localFilePath] = [document, context];
+                const data: string = await this.getData(context);
+                const textEditor: vscode.TextEditor = await vscode.window.showTextDocument(document);
+                await this.updateEditor(data, textEditor);
+                this.appendLineToOutput(' Done.');
+            } catch (error) {
+                let details: string;
+
+                // tslint:disable-next-line:no-unsafe-any
+                if (!!error.message) {
+                    // tslint:disable-next-line:no-unsafe-any
+                    details = error.message;
+                } else {
+                    details = JSON.stringify(error);
+                }
+
+                this.appendLineToOutput(localize('failed', " Failed."));
+                this.appendLineToOutput(localize('errorDetails', 'Error Details: {0}', details));
+
+                await vscode.window.showWarningMessage(localize('downloadError', 'Unable to download "{0}". Please check Output for more information.', fileName), dialogResponses.OK);
+            }
+
+        }
+    }
+
+    public async updateMatchingcontext(doc: vscode.Uri): Promise<void> {

```

Comment:
Fix the camelCase for context

Reply:
@JakeRadMSFT I didn't actually see this function getting called anywhere.  Do we need to include this in the BaseEditor?

Reply:
A similar function is used in Cosmos DB if you have the 'Update to Azure' right click menu or button in the file.

However, I don't think Jake use's it in Storage because saving the file is his only entry point